enhancements/installer/feature-sets.md

```diff
@@ -0,0 +1,209 @@
+to gate Installer features that are in active development.
+
+## Motivation
+
+Feature gates have been standard in OpenShift since at least 4.1, but have been absent from the Installer.
+There are many observable benefits of the practice, but the primary motivations are:
+
+* Align the Installer with OpenShift release practices and graduation criteria, i.e. enable QE & CI testing
+of gated features
+* Improve Installer developer experience by removing concerns about exposing non-GA features during active development
+* Allow a mechanism for pre-release user testing
+
+We have an immediate use case for this as we are targeted to release a tech preview feature in 4.12.
+
+### User Stories
+
+As a member of OpenShift concerned with the release process (TRT, dev, staff engineer, maybe even PM), I want to opt in to
+pre-release features so that I can run periodic testing in CI and obtain a signal of feature quality.  
+
+As quality engineering, I want to opt in to pre-release features so that I can test nightly and other release images 
+both in CI & locally.
+
+As a customer, I want to opt in to pre-release features so that I can test and verify proof of concepts. 
+
+As an Installer developer, I want to protect new API fields with a feature gate so that I can build features without concern
+that an incomplete feature would be exposed to customers.
+
+### Goals
+
+The goals of this proposal are:
+* To provide an API that will allow users to enable feature sets in the Installer
+* Allow the Installer to enable feature sets in the cluster
+* Ensure the proposed implementation adheres to best practices for feature sets
+
+### Non-Goals
+
+None
+
+## Proposal
+
+### Workflow Description
+
+Example workflow demonstrating feature set validation and use:
+
+**Developer** contributes code to the openshift-install codebase.
+
+**Admin** is the user performing an install.
+
+1. Developer adds feature `foo` to openshift-install. Feature `foo`
+includes field `bar` in the install config; feature `foo` is part of the Tech Preview Feature Set.
+2. Admin specifies `bar: baz` in install-config.yaml (but `featureSet` is empty).
+3. `openshift-install` returns an error: `the TechPreviewNoUpgrade feature set must be enabled to use this field`
+4. Admin adds `featureSet: TechPreviewNoUpgrade` to install-config.yaml
+5. Installer generates `FeatureGate` manifest
+6. Cluster installs successfully with TechPreview cluster 
+
+
+### API Extensions
+
+[OpenShift API Feature Sets](https://github.com/openshift/api/blob/master/config/v1/types_feature.go)
+would become part of the install config API:
+
+```go
+
+type FeatureSet configv1.FeatureSet
+
+const (
+	// ExperimentalNoUpgrade enables features that are experimental or in active development.
+	ExperimentalNoUpgrade configv1.FeatureSet = "ExperimentalNoUpgrade"
+)
+```
+
+`ExperimentalNoUpgrade`, for Installer work in progress or
```

Comment:
Ok, so it's good that we're co-locating here.  How would these feature differ from TechPreviewNoUpgrade?  TechPreview allows customers to call support, get our best attempt at fixing a problem, but utlimately we can say: "very sorry, that's just not going to work".

Basically every team has claimed, "but my feature is special and people will want just this one and they don't want to risk running all those other less stable features".  In practice, this has not been a problem and the single switch makes it easier for us to keep a small-ish test matrix for our testing.

Reply:
removed experimentalnoupgrade